examples/microwave-oven-app/microwave-oven-common/include/microwave-oven-device.h

```diff
@@ -0,0 +1,89 @@
+#include <app/clusters/operational-state-server/operational-state-server.h>
+
+namespace chip {
+namespace app {
+namespace Clusters {
+
+class MicrowaveOvenDevice
+{
+private:
+    MicrowaveOvenControl::MicrowaveOvenControlDelegate mMicrowaveOvenControlDelegate;
+    MicrowaveOvenControl::Instance mMicrowaveOvenControlInstance;
+
+    OperationalState::OperationalStateDelegate mOperationalStateDelegate;
+    OperationalState::Instance mOperationalStateInstance;
+
+public:
+    /**
+     * This class is responsible for initialising all the microwave oven device clusters and manging the interactions between them
+     * as required by the specific "business logic". See the state machine diagram.
+     * @param aClustersEndpoint The endpoint ID where all the microwave oven clusters exist.
+     */
+    explicit MicrowaveOvenDevice(EndpointId aClustersEndpoint) :
+        mMicrowaveOvenControlDelegate(),
+        mMicrowaveOvenControlInstance(&mMicrowaveOvenControlDelegate, aClustersEndpoint, MicrowaveOvenControl::Id),
+        mOperationalStateDelegate(), mOperationalStateInstance(&mOperationalStateDelegate, aClustersEndpoint, OperationalState::Id)
+    {
+        // set callback functions
+        mMicrowaveOvenControlDelegate.SetMicrowaveOvenControlSetCookingParametersCallback(
+            &MicrowaveOvenDevice::HandleMicrowaveOvenSetCookingParametersCallback, this);
+        mMicrowaveOvenControlDelegate.SetMicrowaveOvenControlAddMoreTimeCallback(
+            &MicrowaveOvenDevice::HandleMicrowaveOvenAddMoreTimeCallback, this);
+        mOperationalStateDelegate.SetOpStatePauseCallback(&MicrowaveOvenDevice::HandleMicrowaveOvenOpStatePasueCallback, this);
+        mOperationalStateDelegate.SetOpStateResumeCallback(&MicrowaveOvenDevice::HandleMicrowaveOvenOpStateResumeCallback, this);
+        mOperationalStateDelegate.SetOpStateStartCallback(&MicrowaveOvenDevice::HandleMicrowaveOvenOpStateStartCallback, this);
+        mOperationalStateDelegate.SetOpStateStopCallback(&MicrowaveOvenDevice::HandleMicrowaveOvenOpStateStopCallback, this);
+        mOperationalStateDelegate.SetOpStateGetCountdownTimeCallback(
+            &MicrowaveOvenDevice::HandleMicrowaveOvenOpStateGetCountdownTime, this);
+    }
+
+    /**
+     * Init all the clusters used by this device.
+     */
+    void Init(EndpointId aEndpoint);
+
+    /**
+     * handle command for microwave oven control: set cooking parameters
+     */
+    Protocols::InteractionModel::Status HandleMicrowaveOvenSetCookingParametersCallback(uint8_t cookMode, uint32_t cookTime,
+                                                                                        uint8_t powerSetting);
+
+    /**
+     * handle command for microwave oven control: add more time
+     */
+    Protocols::InteractionModel::Status HandleMicrowaveOvenAddMoreTimeCallback(uint32_t addedCookTime);
+
+    /**
+     * handle command for operational state: pause
```

Comment:
Please document what will be set in `err` in different cases, for all these methods.